File: D1/go/main.go
package main

import (
	"encoding/json"
	"io"
	"net/http"

	"github.com/gin-gonic/gin"
)

func main() {
	router := gin.Default()

	router.GET("/ping", func(c *gin.Context) {
		c.JSON(http.StatusOK, gin.H{"message": "Pong!"})
	})

	router.GET("/forward", func(c *gin.Context) {
		url := c.Query("url")
		if url == "" {
			c.JSON(http.StatusBadRequest, gin.H{"message": "Missing url parameter"})
			return
		}

		resp, err := http.Get(url)
		if err != nil {
			c.JSON(http.StatusInternalServerError, gin.H{"message": "Error executing request"})
			return
		}
		defer resp.Body.Close()

		body, err := io.ReadAll(resp.Body)
		if err != nil {
			c.JSON(http.StatusInternalServerError, gin.H{"message": "Error reading response body"})
			return
		}

		var jsonResult map[string]interface{}
		err = json.Unmarshal(body, &jsonResult)
		if err != nil {
			c.JSON(http.StatusInternalServerError, gin.H{"message": "Error unmarshalling JSON"})
			return
		}

		c.JSON(http.StatusOK, jsonResult)
	})

	router.Run(":80")
}

File: D1/node/main.js
const express = require('express');
const axios = require('axios');
const app = express();
const port = 80;

app.get('/ping', (req, res) => {
  res.status(200).json({ message: 'Pong!' });
  console.log('[Node] ' + req.url);
});

app.get('/forward', async (req, res) => {
  const url = req.query.url;

  if (!url) {
    res.status(400).json({ message: 'Missing url parameter' });
    return;
  }

  try {
    const response = await axios.get(url);
    res.status(200).json(response.data);
  } catch (error) {
    res.status(500).json({ message: 'Error executing request' });
  }
  console.log('[Node] ' + req.url);
});

app.listen(port, () => {
  console.log(`Server is running on port ${port}`);
});